src/shared/components/create-case-filters/create-case-filters.component.ts

```diff
@@ -103,14 +104,29 @@ export class CreateCaseFiltersComponent implements OnInit {
     });
   }
 
-  private sortEvents(events: CaseEvent[]) {
-    return this.orderService.sort(this.retainEventsWithNoPreStates(events));
+  private sortEvents(events: CaseEvent[]): CaseEvent[] {
+    return this.orderService.sort(this.retainEventsWithCreateRights(this.retainEventsWithNoPreStates(events)));
   }
 
-  private retainEventsWithNoPreStates(events: CaseEvent[]) {
+  private retainEventsWithNoPreStates(events: CaseEvent[]): CaseEvent[] {
     return events.filter(event => event.pre_states.length === 0);
   }
 
+  private retainEventsWithCreateRights(events: CaseEvent[]): CaseEvent[] {
+    const userProfile = JSON.parse(this.sessionStorageService.getItem('userDetails'));
+    const caseEvents: CaseEvent[] = [];
+    events.forEach((event) => {
```

Comment:
You could easily simplify this code 
`return events.filter(event => userProfile && userProfile.roles && !!userProfile.roles.find(role => this.hasCreateAccess(event, role)) )`

Reply:
updated the code